lib/matrix_convert_to_zero.rb

```diff
@@ -1,10 +1,25 @@
-# Updates the input matrix based on the following rules:
-# Assumption/ Given: All numbers in the matrix are 0s or 1s
-# If any number is found to be 0, the method updates all the numbers in the
-# corresponding row as well as the corresponding column to be 0.
+# Time complexity: O(n*m), where n and m are nums of rows and columns
+# Space complexity: O(n + m), where n and m are nums of rows and columns
```

Comment:
Space complexity looks better than `n + m` to me. 🙂

Any ideas on what it might be?

Reply:
Oh, good question! So ... since this method doesn't create any extra structures to hold data, is it just constant? This was my first try at a matrix and I think I forgot that all it is is a single particularly organized array. O(n)?

Reply:
I'd agree with the former - if you wanted to be super-technically correct, you could say "constant _additional_ space". (Excluding the size of your input and output variable(s) is [a matter of convention](https://cs.stackexchange.com/questions/83574/does-space-complexity-analysis-usually-include-output-space)).

Reply:
I'd agree with the former - if you wanted to be super-technically correct, you could say "constant _additional_ space". (Excluding the size of your input and output variable(s) is [essentially a matter of convention](https://cs.stackexchange.com/questions/83574/does-space-complexity-analysis-usually-include-output-space)).